### PR TITLE
ref(ts) Update project general settings to typescript

### PR DESCRIPTION
**Pull request by markstory**

### CODE REVIEW

File: src/sentry/static/sentry/app/views/settings/projectGeneralSettings.tsx [hunk 1]
```diff
@@ -77,7 +78,7 @@ class ProjectGeneralSettings extends AsyncView {
     if (!project) {
       return;
     }
-    if (!this._form.email) {
+    if (!this._form.email || typeof this._form.email !== 'string') {
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/settings/projectGeneralSettings.tsx:81]: is `!this._form.email` redundant?

**Review by billyvg — APPROVED** [review]

File: src/sentry/static/sentry/app/views/settings/projectGeneralSettings.tsx [hunk 2]
```diff
@@ -220,7 +219,7 @@ class ProjectGeneralSettings extends AsyncView {
   }
 
   renderBody() {
-    const {organization} = this.context;
+    const {organization} = this.props;
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/settings/projectGeneralSettings.tsx:222]: 👏

File: src/sentry/static/sentry/app/views/settings/projectGeneralSettings.tsx [hunk 3]
```diff
@@ -233,6 +232,7 @@ class ProjectGeneralSettings extends AsyncView {
       access,
       disabled: !access.has('project:write'),
     };
+    const team = project.teams.length ? project.teams[0] : undefined;
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/settings/projectGeneralSettings.tsx:235]:
```suggestion
    const team = project.teams?.[0];
```

**Review by priscilawebdev — APPROVED** [review]
I agree with @billyvg suggestion, but other than that it lgtm 👍

File: src/sentry/static/sentry/app/views/settings/projectGeneralSettings.tsx (hunk 1 above)
Comment (markstory) [reply src/sentry/static/sentry/app/views/settings/projectGeneralSettings.tsx:81]: It would still keep '' out.

**Pull request merged**